src/components/PeoplePage.tsx

```diff
@@ -1,8 +1,88 @@
-import { PeopleFilters } from './PeopleFilters';
+import { useEffect, useMemo, useState } from 'react';
+import { useSearchParams } from 'react-router-dom';
+import { getPeople } from '../api';
+import { Person } from '../types';
 import { Loader } from './Loader';
+import { PeopleFilters } from './PeopleFilters';
 import { PeopleTable } from './PeopleTable';
 
 export const PeoplePage = () => {
+  const [people, setPeople] = useState<Person[]>([]);
+  const [isLoading, setIsLoading] = useState<boolean>(true);
+  const [error, setError] = useState(false);
+
+  const [searchParams] = useSearchParams();
+
+  const sexParam = searchParams.get('sex') || 'all';
+  const query = searchParams.get('query');
+  const centuries = searchParams.getAll('centuries') || [];
+
+  const isEmptyFilter = useMemo(() => {
+    return !searchParams.toString();
+  }, [searchParams]);
+
+  const handleGetPeople = async () => {
+    try {
+      const data = await getPeople();
+
+      const dataWithParents = data.map(person => {
+        return {
+          ...person,
+          mother: data.find(anyone => anyone.name === person.motherName),
+          father: data.find(anyone => anyone.name === person.fatherName),
+        };
+      });
+
+      setPeople(dataWithParents);
+    } catch {
+      setError(true);
+    } finally {
+      setIsLoading(false);
+    }
+  };
+
+  const handleFilterPeople = () => {
+    let newPeope: Person[] = [];
+
+    if (sexParam !== 'all') {
+      newPeope = people.filter((person: Person) => person.sex === sexParam);
+    } else {
+      newPeope = [...people];
+    }
+
+    if (query) {
+      newPeope = newPeope.filter((person: Person) => {
+        const queryLowerCase = query.toLowerCase();
+
+        return person.name.toLowerCase().includes(queryLowerCase)
```

Comment:
1. destructure person
2. `toLowerCase().includes(queryLowerCase)` - repeated many times. Move to helper function and set name and query to params. For exmaple:
`const { name,  fatherName, motherName } = person`
....

`myAwesomeFunction(name, query) || myAwesomeFunction(fatherName, query) || myAwesomeFunction(motherName, query)`